app/models/validators/range_overlapping_validator.rb

```diff
@@ -1,5 +1,7 @@
 class RangeOverlappingValidator < ActiveModel::Validator
   def validate(record)
+    return if record.metric_configuration.nil?
```

Comment:
I actually think the other condition would be better in the validator itself. I hate writing code as strings, and possibly having to repeat a pre-condition that can cause an exception (not just bad results) everywhere.

You're right that I missed a test, I actually did a quick fix since it broke the seeds and forgot to clean it up.